libmamba/src/core/menuinst.cpp

```diff
@@ -333,56 +339,85 @@ namespace mamba
-                    script = tmp[0];
-                    if (tmp.size() > 1)
+                    name = item["name"];  // Should be a string
+
+                    if (item.contains("pywscript"))
                     {
-                        std::copy(tmp.begin() + 1, tmp.end(), back_inserter(arguments));
+                        script = root_pyw;
+                        arguments = cwp_pyw_args;
+                        auto tmp = util::split(item["pywscript"], " ");
+                        std::copy(tmp.begin(), tmp.end(), back_inserter(arguments));
+                    }
+                    else if (item.contains("pyscript"))
+                    {
+                        script = root_py;
+                        arguments = cwp_py_args;
+                        auto tmp = util::split(item["pyscript"], " ");
+                        std::copy(tmp.begin(), tmp.end(), back_inserter(arguments));
+                    }
+                    else if (item.contains("webbrowser"))
+                    {
+                        script = root_pyw;
+                        arguments = { "-m", "webbrowser", "-t", item["webbrowser"] };
+                    }
+                    else if (item.contains("script"))
+                    {
+                        script = root_py;
+                        arguments = { cwp_path.string(), target_prefix.string() };
+                        auto tmp = util::split(item["script"], " ");
+                        std::copy(tmp.begin(), tmp.end(), back_inserter(arguments));
+                        extend_script_args(item, arguments);
+                    }
+                    else if (item.contains("system"))
+                    {
+                        auto tmp = util::split(item["system"], " ");
+                        script = tmp[0];
+                        if (tmp.size() > 1)
+                        {
+                            std::copy(tmp.begin() + 1, tmp.end(), back_inserter(arguments));
+                        }
+                        extend_script_args(item, arguments);
+                    }
+                    else
+                    {
+                        LOG_ERROR << "Unknown shortcut type found in " << json_file;
+                        throw std::runtime_error("Unknown shortcut type.");
                     }
-                    extend_script_args(item, arguments);
                 }
-                else
+                else  // MenuInstVersion::Version2
                 {
-                    LOG_ERROR << "Unknown shortcut type found in " << json_file;
-                    throw std::runtime_error("Unknown shortcut type.");
+                    // `item["name"]` should be an object containing items with
+                    // "target_environment_is_base" and "target_environment_is_not_base"(default)
+                    // as keys
+                    name = item["name"]["target_environment_is_not_base"];
```

Comment:
This is not totally correct. It could also be a string. See schema at https://conda.github.io/menuinst/reference/#menuinst._schema.MenuItem.name.